src/MultiMessageReceiver.sol

```diff
@@ -199,16 +199,34 @@ contract MultiMessageReceiver is IMultiMessageReceiver, ExecutorAware, Initializ
         }
     }
 
-    /// @notice Update power quorum threshold of message execution.
-    function updateQuorum(uint64 _quorum) external onlyGovernanceTimelock {
-        /// NOTE: should check 2/3 ?
-        if (_quorum > trustedExecutor.length || _quorum == 0) {
-            revert Error.INVALID_QUORUM_THRESHOLD();
+    /// @notice Update bridge receiver adapters after quorum update
+    /// @dev called by admin to update receiver bridge adapters on all other chains
+    function updateQuorumAndReceiverAdapter(
+        uint64 _newQuorum,
+        address[] calldata _receiverAdapters,
+        bool[] calldata _operations
+    ) external onlyGovernanceTimelock {
+        uint256 len = _receiverAdapters.length;
+
+        if (len != _operations.length) {
+            revert Error.ARRAY_LENGTH_MISMATCHED();
         }
-        uint64 oldValue = quorum;
 
-        quorum = _quorum;
-        emit QuorumUpdated(oldValue, _quorum);
+        /// @dev updates quorum here
+        _updateQuorum(_newQuorum);
+
+        for (uint256 i; i < len;) {
+            _updateReceiverAdapter(_receiverAdapters[i], _operations[i]);
+
+            unchecked {
+                ++i;
+            }
+        }
```

Comment:
suggest defining an internal `_updateReceiverAdapters(address[] _adapters, bool[] _operations)` function and calling it here. 
We can call that function from the external `updateReceiverAdapter(address[]...)` function (line 183). We should also probably rename that function to its plural form given it updates multiple adapters.